tests/kafkatest/services/kafka/kafka.py

```diff
@@ -323,8 +326,12 @@ def start_cmd(self, node):
         cmd += "export KAFKA_LOG4J_OPTS=\"-Dlog4j.configuration=file:%s\"; " % self.LOG4J_CONFIG
         heap_kafka_opts = "-XX:+HeapDumpOnOutOfMemoryError -XX:HeapDumpPath=%s" % \
                           self.logs["kafka_heap_dump_file"]["path"]
+        # https://docs.confluent.io/current/kafka/deployment.html#jvm
+        prod_kafka_opts = ("-Xms{0} -Xmx{0} -XX:MetaspaceSize=96m -XX:+UseG1GC -XX:MaxGCPauseMillis=20 ".format(self.jvm_heap_size)
+                           + "-XX:InitiatingHeapOccupancyPercent=35 -XX:G1HeapRegionSize=16M "
+                           + "-XX:MinMetaspaceFreeRatio=50 -XX:MaxMetaspaceFreeRatio=80")
```

Comment:
Is there evidence that we really need all this stuff?  It will just be another thing that we need to change when we move to a new version of the JDK.  I think a better approach would be to take this stuff from a local file or an environment variable passed to the ducktape process.

Reply:
> Is there evidence that we really need all this stuff?
well my main goal in this specific downstream test is to replicate a production environment and since we recommend that users to tweak these in production, I think it makes sense to include them (regardless of the whether or not they're essential)

>  It will just be another thing that we need to change when we move to a new version of the JDK.
this is true. there may be some overhead here... 🤔

> I think a better approach would be to take this stuff from a local file or an environment variable passed to the ducktape process.
what if we just allowed users to pass in their own additions to `KAFKA_OPTS` via service constructor? something like `self.extra_kafka_opts`?